### ISSUE TITLE
TypeError when Chardet found encoding None

### ISSUE THREAD

**Issue by chiunam** [Diaoul/subliminal#921]
Subliminal sometimes fallback to use Chardet to guess the encoding of downloaded subtitles. If Chardet could not determine the encoding and return `None`, error thrown `TypeError: decode() argument 1 must be string, not None`

### command
```bash
subliminal --debug -v -l zh -l en -p shooter -p addic7ed -p opensubtitles -a 12w
```

### debug log
```log
...
INFO:subliminal.providers.shooter:Downloading subtitle <ShooterSubtitle u'https://www.shooter.cn/api/subapi.php?fetch=MTUyNzIxNDI5NXw4RkF6T2JWZHlydElodExUSlRfVmtQdmY3Z1FlOGQ3Tk5rUWVpcUc3WFdWNWRwM1NNZWNnUGVDSG1FT0M3SE1BNUhXVWQwQlVIaVJ2RXhLYUhoMjdPS1RJQ18wSjdYejA3MHFGVkp3dmdqV0JoaFBkc1dZSE9HdEtYWm83LTV2clNaVnRIN0E2YnUwb1hqbGZwbEhIdnJQVmJ4SzR2bkE9fMZBkGDT5kPh1pE11y7bmlqUTdQGfcOmc8wTr-rR1uRd&nonce=%86r%FB%F9%24%A9%27%96%00u%AF%C1%C0%C5%9C%A1' [zh]>
INFO:subliminal.subtitle:Guessing encoding for language zh
DEBUG:subliminal.subtitle:Trying encodings ['utf-8', 'gb18030', 'big5']
WARNING:subliminal.subtitle:Could not guess encoding from language
INFO:subliminal.subtitle:Chardet found encoding None
DEBUG:subliminal.core:Terminating initialized providers
INFO:subliminal.core:Terminating provider shooter
Traceback (most recent call last):
  File "/opt/bin/subliminal", line 11, in <module>
    sys.exit(subliminal())
  File "/opt/lib/python2.7/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/opt/lib/python2.7/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/opt/lib/python2.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/opt/lib/python2.7/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/opt/lib/python2.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/opt/lib/python2.7/site-packages/click/decorators.py", line 27, in new_func
    return f(get_current_context().obj, *args, **kwargs)
  File "/opt/lib/python2.7/site-packages/subliminal/cli.py", line 403, in download
    hearing_impaired=hearing_impaired, only_one=single)
  File "/opt/lib/python2.7/site-packages/subliminal/core.py", line 227, in download_best_subtitles
    if self.download_subtitle(subtitle):
  File "/opt/lib/python2.7/site-packages/subliminal/core.py", line 182, in download_subtitle
    if not subtitle.is_valid():
  File "/opt/lib/python2.7/site-packages/subliminal/subtitle.py", line 85, in is_valid
    if not self.text:
  File "/opt/lib/python2.7/site-packages/subliminal/subtitle.py", line 76, in text
    return self.content.decode(self.guess_encoding(), errors='replace')
TypeError: decode() argument 1 must be string, not None

```

**Comment by ratoaq2** [comment]
I would like to create a test case with this scenario but I'm still not able to have an input that chardet returns None as the detected encoding

**Comment by ratoaq2** [comment]
Fixed by 78d8f26fdd5c136d264033859ff92c1b86ee8727
